docs/snippets/package/aws-with-vpc/crossplane.yaml

```diff
@@ -11,4 +26,4 @@ spec:
     version: ">=v1.4.0-0"
   dependsOn:
     - provider: crossplane/provider-aws
-      version: ">=v0.18.2"
```

Comment:
latest aws provider is 0.30.1 - is there any reason not to use that version?  Driving users (especially new users) to later versions may save them time and aggravation when bringing things up for the first time.

Reply:
it's not pinned, so we will still install the latest available versions from `docker.io/crossplane/...`:

```
➜  crossplane git:(jastang-update-xpkg-metadata) k get providers
NAME                        INSTALLED   HEALTHY   PACKAGE                             AGE
crossplane-provider-aws     True        True      crossplane/provider-aws:v0.30.1     2d3h
crossplane-provider-azure   True        True      crossplane/provider-azure:v0.19.0   47h
crossplane-provider-gcp     True        True      crossplane/provider-gcp:v0.21.0     47h
```

there's some work to simultaneously publish to Docker hub and Upbound registries, for example [here](https://github.com/crossplane-contrib/provider-gcp/issues/453), and by default the `dependsOn.provider` looks at the Upbound registry so I'm using the latest available version there. once the backfill is done and everything is in sync we can bump these up :)

Reply:
General philosophy on error constraints in Crossplane packages: usually a `>=` with the earliest known compatible version (e.g. MSV - Minimum Supported Version) is a good policy (both for the package and Crossplane version constraints). Otherwise, you are potentially excluding someone with an existing install of dependencies (`provider-aws` in this case) or Crossplane from using your package when in reality it is compatible. If the user does not have the dependencies installed, Crossplane is going to resolve to whatever the latest version is at that point (as @jastang mentions).

Reply:
To avoid just providing feedback without any suggested action: I think the bumps as currently proposed in this PR are still okay given that these getting started packages are tagged alongside their Crossplane versions, and `v0.24.1` has been out for a sufficiently long period of time to not break most folks, and these packages are primarily used for new installs anyway 👍🏻

Reply:
Sounds good to me - thanks @hasheddan !